apps/sms/views/conversation/js/conversation.js

```diff
@@ -2014,13 +2036,25 @@ var ConversationView = {
     ).then(() => {
       WaitingScreen.show();
 
-      var messageIdsToDelete = this.selectionHandler.selectedList.map(
-        (item) => +item
-      );
-
-      return this.deleteMessages(messageIdsToDelete);
+      if(ConversationView.inEditMode) {
+        messageIdsToDelete = this.selectionHandler.selectedList.map(
+          (item) => +item
+        );
+        return onAllThreadMessagesRetrieved(messageIdsToDelete);
+      } else {
+        console.log('test1');
+        MessageManager.getMessages({
+        // Filter and request all messages with this threadId
+          filter: { threadId: ConversationView.activeThread.id },
+          each: onThreadMessageRetrieved,
+          end: onAllThreadMessagesRetrieved
```

Comment:
Mmm, I think we should not call ```deleteMessages``` for the case when we delete entire thread - it does too much useless work, let's have ```deleteConversation``` method that will be called here instead.

And please use ```this.``` instead of ```ConversationView.```` where possible.